### PR TITLE
Fixing Android tests and add them to CI

### PR DESCRIPTION
**Pull request by shridharkalagi**
## Change list
1. Fixing the android tests 
2. Add Android CI integration

## Types of changes

What types of changes are you proposing/introducing to Java client?
_Put an `x` in the boxes that apply_

- [ ] No changes in production code.
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Details

Have Included few tests in CI which always pass on the latest version on Android Emulator available on travis (android-25)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,33 +13,43 @@ matrix:
...
+        - echo y | android update sdk -u -a -t extra-android-m2repository
+      os: linux
+      jdk: oraclejdk8
+      android:
+        components:
+          - extra-android-support
+          - sys-img-armeabi-v7a-android-24
+      before_install:
+        - curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.1/install.sh | bash
+        - nvm install 6
```
Comment (mykola-mokhnach) [inline .travis.yml]: I assume the minimum node version should be 8

File: src/test/java/io/appium/java_client/android/AndroidDriverTest.java [hunk 2]
```diff
@@ -209,8 +209,9 @@
     }
 
     @Test public void pullFileTest() {
+        //Works only on rooted devices/Emulator and adb is running as root
```
Comment (mykola-mokhnach) [inline src/test/java/io/appium/java_client/android/AndroidDriverTest.java]: this actually depends on the actual path. For example the content of `/sdcard` is available without root access

File: src/test/java/io/appium/java_client/android/AndroidFunctionTest.java [hunk 3]
```diff
@@ -44,7 +44,8 @@
             if (m.find()) {
                 driver.context(context);
             }
-        });
```
Comment (mykola-mokhnach) [inline src/test/java/io/appium/java_client/android/AndroidFunctionTest.java]: is this change needed?

File: src/test/java/io/appium/java_client/android/AndroidLogcatListenerTest.java [hunk 4]
```diff
@@ -12,7 +12,7 @@
 public class AndroidLogcatListenerTest extends BaseAndroidTest {
 
     @Test
-    public void verifyLogcatListenerCanBeAssigned() {
+    public void verifyLogcatListenerCanBeAssigned() {
```
Comment (mykola-mokhnach) [inline src/test/java/io/appium/java_client/android/AndroidLogcatListenerTest.java]: is this change needed?

File: src/test/java/io/appium/java_client/android/FingerPrintTest.java [hunk 5]
```diff
@@ -94,26 +102,29 @@ private void clickOnSecurity() {
...
+    @Before
+    public void before() {
         initDriver();
         clickOnSecurity();
         findElementByText("Screen lock").click();
         findElementByText("PIN").click();
         enterPasswordAndContinue();
         enterPasswordAndContinue();
-        clickNext();
+        //clickNext();
```
Comment (mykola-mokhnach) [inline src/test/java/io/appium/java_client/android/FingerPrintTest.java]: why the method call is commented?

File: src/main/java/io/appium/java_client/Setting.java [hunk 6]
```diff
@@ -23,10 +23,10 @@
...
     IGNORE_UNIMPORTANT_VIEWS("ignoreUnimportantViews"),
     WAIT_FOR_IDLE_TIMEOUT("waitForIdleTimeout"),
-    WAIT_FOR_SELECTOR_TIMEOUT("setWaitForSelectorTimeout"),
-    WAIT_SCROLL_ACKNOWLEDGMENT_TIMEOUT("setScrollAcknowledgmentTimeout"),
-    WAIT_ACTION_ACKNOWLEDGMENT_TIMEOUT("setActionAcknowledgmentTimeout"),
-    KEY_INJECTION_DELAY("setKeyInjectionDelay"),
+    WAIT_FOR_SELECTOR_TIMEOUT("waitForSelectorTimeout"),
+    WAIT_SCROLL_ACKNOWLEDGMENT_TIMEOUT("scrollAcknowledgmentTimeout"),
+    WAIT_ACTION_ACKNOWLEDGMENT_TIMEOUT("actionAcknowledgmentTimeout"),
+    KEY_INJECTION_DELAY("keyInjectionDelay"),
```
Comment (SrinivasanTarget) [inline src/main/java/io/appium/java_client/Setting.java]: Is this changed intended?

Comment (shridharkalagi) [reply src/main/java/io/appium/java_client/Setting.java]: yes @SrinivasanTarget  its intended

**Comment by saikrishna321** [comment]
This PR has got in some of the previous commits

File: src/main/java/io/appium/java_client/Setting.java (hunk 6 above)
Comment (SrinivasanTarget) [reply src/main/java/io/appium/java_client/Setting.java]: OK I could see the changes [here](https://github.com/appium/appium-uiautomator2-server/pull/138/files)
